nautobot/extras/factory.py

```diff
@@ -158,6 +159,22 @@ class Params:
     tenant = factory.Maybe("has_tenant", random_instance(Tenant))
 
 
+class JobQueueAssignmentFactory(BaseModelFactory):
+    """JobQueueAssignment Factory"""
+
+    class Meta:
+        model = JobQueueAssignment
+
+    job = random_instance(Job, allow_null=True)
+
+    @factory.lazy_attribute
+    def job_queue(self):
+        job_queue = factory.random.randgen.choice(JobQueue.objects.all())
+        while JobQueueAssignment.objects.filter(job=self.job, job_queue=job_queue).exists():
+            job_queue = factory.random.randgen.choice(JobQueue.objects.all())
+        return job_queue
```

Comment:
Possibility of an infinite loop here if a job is already assigned to all queues.

As an alternative, did you consider having `JobQueueFactory` automatically assign jobs to the queues rather than using a separate factory?

Reply:
Refactored it and remove JobQueueAssignmentFactory